lib/_stream_readable.js

```diff
@@ -522,8 +524,13 @@ Readable.prototype.pipe = function(dest, pipeOpts) {
     if (false === ret) {
       debug('false write response, pause',
             src._readableState.awaitDrain);
-      src._readableState.awaitDrain++;
-      src.pause();
+      // If the user unpiped during `dest.write()`, it is possible
+      // to get stuck in a permanently paused state if that write
+      // also failed.
+      if (!cleanedUp) {
+        src._readableState.awaitDrain++;
+        src.pause();
```

Comment:
I think the desired behavior would be:

1. If `src` is only piped to `dst` & has no other `"data"` listeners, `pause` `src` without increasing `awaitDrain`.
2. do not pause `src` if:
  1. `src` has more than one `dst`
  2. `src` has `"data"` listeners that were added with `Readable#on` (ones that trigger flowing mode)